sdk/core/Azure.Core/src/Pipeline/Internal/ReadTimeoutStream.cs

```diff
@@ -138,7 +138,10 @@ private void UpdateReadTimeout()
         {
             try
```

Comment:
Do you still need the try/catch if you are checking `CanTimeout`?

Reply:
I wanted to keep it for some possible badly-behaved streams.